packages/kbn-bazel-packages/src/discover_packages.ts

```diff
@@ -9,23 +9,46 @@
 import Path from 'path';
 
 import globby from 'globby';
+import normalizePath from 'normalize-path';
 import { REPO_ROOT } from '@kbn/utils';
 import { asyncMapWithLimit } from '@kbn/std';
 
 import { BazelPackage } from './bazel_package';
 import { BAZEL_PACKAGE_DIRS } from './bazel_package_dirs';
 
 export function discoverBazelPackageLocations(repoRoot: string) {
-  return globby
+  const packagesWithPackageJson = globby
     .sync(
       BAZEL_PACKAGE_DIRS.map((dir) => `${dir}/*/package.json`),
       {
         cwd: repoRoot,
         absolute: true,
       }
     )
+    // NOTE: removing x-pack by default for now to prevent a situation where a BUILD.bazel file
+    // needs to be added at the root of the folder which will make x-pack to be wrongly recognized
+    // as a Bazel package in that case
+    .filter((path) => !normalizePath(path).includes('x-pack/package.json'))
     .sort((a, b) => a.localeCompare(b))
     .map((path) => Path.dirname(path));
+
+  const packagesWithBuildBazel = globby
+    .sync(
+      BAZEL_PACKAGE_DIRS.map((dir) => `${dir}/*/BUILD.bazel`),
+      {
+        cwd: repoRoot,
+        absolute: true,
+      }
+    )
+    .map((path) => Path.dirname(path))
+    .reduce((accum: { [key: string]: boolean }, curr: string) => {
+      accum[curr] = true;
+      return accum;
+    }, {});
```

Comment:
Can't we just use `array.includes()` instead of turning this into a map?

Reply:
We can but won't it be better to use a map to check it? Probably with the limited amount of packages we will have that won't matter anyway.

Reply:
done

Reply:
I would definitely be convinced that this is better, it seems like a list of things we have and we want to check for membership in that list. I could see an argument for turning it into a `Set()` for the performance of `Set#has()` over `Array#includes()` or something, but I don't think there's a good reason to do this here. Seems totally unexpected.